File: Data Structures & Algorithms/house-robber/submission-3.py
class Solution:
    def rob(self, nums: List[int]) -> int:

        if not nums:
            return 0
        if len(nums) == 1:
            return nums[0]

        DP = [0] * (len(nums))
        DP[0] = nums[0]
        DP[1] = max(nums[0],nums[1])

        for i in range(2,len(nums)):
            DP[i] = max(DP[i-1],DP[i-2]+nums[i])
        
        return DP[-1]

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int l = 0;
        int res = 0;

        unordered_set<char> seen;

        for (int r = 0; r < s.size(); r++){
            while (seen.contains(s[r])){
                seen.erase(s[l]);
                l++;
            }
            seen.insert(s[r]);
            res = max(res,r-l+1);
        }

        return res;

    }
};
